sklearn/_isotonic.pyx

```diff
@@ -63,9 +63,7 @@ def _inplace_contiguous_isotonic_regression(floating[::1] y, floating[::1] w):
             i = k
 
 
-def _make_unique(np.ndarray[dtype=floating] X,
-                 np.ndarray[dtype=floating] y,
-                 np.ndarray[dtype=floating] sample_weights):
+def _make_unique(floating [:] X, floating [:] y, floating [:] sample_weights):
```

Comment:
Are we guaranteed to never get read only arrays here then cf https://github.com/scikit-learn/scikit-learn/issues/10624#issuecomment-449560991 ?

Reply:
I misinterpreted the comment. Time to revert this change..